channels/rdpecam/client/camera.h

```diff
@@ -101,7 +101,10 @@ typedef struct
 	CAM_MEDIA_TYPE_DESCRIPTION currMediaType;
 
 	GENERIC_CHANNEL_CALLBACK* hSampleReqChannel;
-	INT nSampleCredits;
+	CRITICAL_SECTION lock;
+	volatile BOOL sampleRequested;
```

Comment:
This should be a queue instead of a boolean. If for whatever reason the server sends two SampleRequests in a short(er) interval (could also be due to some sudden network delay, so does not necessarily have to be a server issue), then only one of them is answered instead of both.
In such case, the second request will then later be handled when we get the frame from the camera device, but that should be fine and still protocol-conform.